backend/views/syson-diagram-statetransition-view/src/main/java/org/eclipse/syson/diagram/statetransition/view/nodes/CompartmentNodeDescriptionProvider.java

```diff
@@ -39,10 +39,10 @@ public CompartmentNodeDescriptionProvider(EClass eClass, EReference eReference,
     protected List<NodeDescription> getDroppableNodes(IViewDiagramElementFinder cache) {
         var acceptedNodeTypes = new ArrayList<NodeDescription>();
 
-        StateTransitionViewDiagramDescriptionProvider.COMPARTMENTS_WITH_LIST_ITEMS.forEach((type, listItems) -> {
-            listItems.forEach(ref -> {
-                if (this.eReference.getEType().equals(ref.getEType())) {
-                    var optCompartmentItemNodeDescription = cache.getNodeDescription(this.descriptionNameGenerator.getCompartmentItemName(type, ref));
+        StateTransitionViewDiagramDescriptionProvider.COMPARTMENTS_WITH_MERGED_LIST_ITEMS.forEach((type, listItems) -> {
+            listItems.stream().findFirst().ifPresent(eReference -> {
```

Comment:
My previous remark for `stream().findFirst()` also applies here and for every other similar cases in this PR